src/policy/fees.cpp

```diff
@@ -998,20 +998,38 @@ void CBlockPolicyEstimator::FlushUnconfirmed() {
     LogPrint(BCLog::ESTIMATEFEE, "Recorded %u unconfirmed txs from mempool in %gs\n", num_entries, (endclear - startclear)*0.000001);
 }
 
-FeeFilterRounder::FeeFilterRounder(const CFeeRate& minIncrementalFee)
+static std::set<double> MakeFeeSet(const CFeeRate& min_incremental_fee,
+                                   double max_filter_fee_rate,
+                                   double fee_filter_spacing)
 {
-    CAmount minFeeLimit = std::max(CAmount(1), minIncrementalFee.GetFeePerK() / 2);
-    feeset.insert(0);
-    for (double bucketBoundary = minFeeLimit; bucketBoundary <= MAX_FILTER_FEERATE; bucketBoundary *= FEE_FILTER_SPACING) {
-        feeset.insert(bucketBoundary);
+    std::set<double> fee_set;
+
+    const CAmount min_fee_limit{std::max(CAmount(1), min_incremental_fee.GetFeePerK() / 2)};
+    fee_set.insert(0);
+    for (double bucket_boundary = min_fee_limit;
+         bucket_boundary <= max_filter_fee_rate;
+         bucket_boundary *= fee_filter_spacing) {
+
+        fee_set.insert(bucket_boundary);
     }
+
+    return fee_set;
+}
+
+FeeFilterRounder::FeeFilterRounder(const CFeeRate& minIncrementalFee)
+    : m_fee_set{MakeFeeSet(minIncrementalFee, MAX_FILTER_FEERATE, FEE_FILTER_SPACING)}
+{
 }
 
 CAmount FeeFilterRounder::round(CAmount currentMinFee)
 {
-    std::set<double>::iterator it = feeset.lower_bound(currentMinFee);
-    if ((it != feeset.begin() && insecure_rand.rand32() % 3 != 0) || it == feeset.end()) {
-        it--;
+    std::set<double>::iterator it = m_fee_set.lower_bound(currentMinFee);
+    const auto PreferLower = [this]() -> bool {
+        LOCK(m_insecure_rand_mutex);
+        return insecure_rand.rand32() % 3 != 0;
+    };
+    if (it == m_fee_set.end() || (it != m_fee_set.begin() && PreferLower())) {
```

Comment:
(This alternative is more straightforward to my eye. Feel free to ignore.)
```diff
-    const auto PreferLower = [this]() -> bool {
-        LOCK(m_insecure_rand_mutex);
-        return insecure_rand.rand32() % 3 != 0;
-    };
-    if (it == m_fee_set.end() || (it != m_fee_set.begin() && PreferLower())) {
+    if (it == m_fee_set.end() || (it != m_fee_set.begin() && WITH_LOCK(m_insecure_rand_mutex, return insecure_rand.rand32()) % 3) {
```

Reply:
Legally help for the safety of worlds

Reply:
You already have a class just for this logic; seems weird to use a lambda when you could just have a dedicated method? (Having a separate function makes the `if` line easier to read IMO -- the github comment already requires scrolling) *shrug*

Reply:
Alright, dropped the function and used `WITH_LOCK()`. I wrapped the line coz long lines arehardtoreadandrequirescrolling in someenvironments (e.g.github) and also when a single piece of it is changed, then the whole line appears as changed in diffs - see the diff at the bottom of https://github.com/bitcoin/bitcoin/pull/15936#discussion_r850190114 for example.

In other words, I agree it is better to be purple, but I like the blueish tint.

Reply:
Making review more difficult on the platform of a trusted third party could be considered a feature :smiling_imp: